src/journal/journalctl.c

```diff
@@ -1452,23 +1458,36 @@ static int list_boots(sd_journal *j) {
         if (count == 0)
                 return count;
 
-        pager_open(arg_pager_flags);
+        table = table_new("#", "boot id", "first entry", "last entry");
+        if (!table)
+                return log_oom();
+
+        if (arg_full)
+                table_set_width(table, 0);
 
         /* numbers are one less, but we need an extra char for the sign */
         w = DECIMAL_STR_WIDTH(count - 1) + 1;
 
         i = 0;
         LIST_FOREACH(boot_list, id, all_ids) {
-                char a[FORMAT_TIMESTAMP_MAX], b[FORMAT_TIMESTAMP_MAX];
-
-                printf("% *i " SD_ID128_FORMAT_STR " %s—%s\n",
-                       w, i - count + 1,
-                       SD_ID128_FORMAT_VAL(id->id),
-                       format_timestamp_maybe_utc(a, sizeof(a), id->first),
-                       format_timestamp_maybe_utc(b, sizeof(b), id->last));
+                char idx[w + 1];
+
+                /* Right-align the index unless JSON is requested */
+                xsprintf(idx, "%*i", OUTPUT_MODE_IS_JSON(arg_output) ? 0 : w, i - count + 1);
```

Comment:
Please use TABLE_INT for `i`, and use `TABLE_SET_ALIGN_PERCENT` to set the alignment. Then, the alignment is gracefully ignored when the output mode is json.

Reply:
Ah, sweet, that's exactly something I was looking for when I was dealing with the alignment, but couldn't find anything relevant.